packages/workspace/src/workspace/workspace.ts

```diff
@@ -1185,18 +1185,23 @@ export const loadWorkspace = async (
   const getElemIdsByStaticFilePaths = async (
     filePaths?: Set<string>,
     envName?: string,
-  ): Promise<Record<string, string>> => {
+  ): Promise<Record<string, string[]>> => {
     const env = envName ?? currentEnv()
     const currentWorkspaceState = await getWorkspaceState()
-    return Object.fromEntries(
-      await awu(currentWorkspaceState.states[env].referencedStaticFiles.entries())
-        .flatMap(({ key: id, value: fileNames }) =>
-          fileNames
-            .filter(filename => filePaths === undefined || filePaths.has(filename))
-            .map(filename => [filename, id]),
-        )
-        .toArray(),
-    )
+    const result: Record<string, string[]> = {}
+    await awu(currentWorkspaceState.states[env].referencedStaticFiles.entries())
+      .flatMap(({ key: id, value: fileNames }) =>
+        fileNames
+          .filter(filename => filePaths === undefined || filePaths.has(filename))
+          .map(filename => [filename, id] as [string, string]),
+      )
+      .forEach(([filename, id]) => {
+        if (!result[filename]) {
+          result[filename] = []
+        }
+        result[filename].push(id)
+      })
+    return result
```

Comment:
convention nit: prefer code without side effects
```suggestion
    const idsAndFileByFileName = await awu(currentWorkspaceState.states[env].referencedStaticFiles.entries())
      .flatMap(({ key: id, value: fileNames }) =>
        fileNames
          .filter(filename => filePaths === undefined || filePaths.has(filename))
          .map(filename => ({ filename, id })),
      )
      .groupBy(({ filename }) => filename)

    return _.mapValues(idsAndFileByFileName, idAndFiles => idAndFiles.map(({ id }) => id))
```